types/react/test/tsx.tsx

```diff
@@ -75,6 +75,9 @@ const StatelessComponentWithoutProps: React.SFC = (props) => {
 };
 <StatelessComponentWithoutProps />;
 
+// React.createContext
+const ContextWithRenderProps = React.createContext('defaultValue');
```

Comment:
I think this is the way too simple test to cover the changes. I would propose following, but I am not entirely sure how to run these tests, so there might some mistake (or several)

```tsx
interface ContextTest = {
  assert: boolean
}
const ContextWithValue = React.createContext<ContextTest>({ assert: true })
const ContextWithoutValue = React.createContext<ContextTest>()

<ContextWithValue.Context>{value => value.assert}</ContextWithValue.Context>

<ContextWithoutValue.Provider value={{ assert: true }}>
  <ContextWithoutValue.Context>{value => value.assert}</ContextWithoutValue.Context>
</ContextWithoutValue.Provider>
```

Reply:
@FredyC I'm not sure myself. The test files here don't seem to be assertion-style tests so much as a sandbox to prove that the typings work without syntax errors. 

I agree that there could be more code here exercising both context scenarios (with and without defaults), but I don't know that they should be executable as none of the rest of the code in this file seems to be.

Reply:
Well yes, tests code is not executed, because you're testing types, not React's behavior.
But tests shouldn't always be positive, if you think that there might be some usage that is not correct, but could be assumed as correct one.

What you have to do for failing tests is add `// $ExpectError` line immediately before the test line. More about that in DT readme:
https://github.com/DefinitelyTyped/DefinitelyTyped#lint 
And dtslint readme:
https://github.com/Microsoft/dtslint#write-tests

Reply:
@scally Are you willing to try that? Perhaps it's overkill for such relatively simple thing. It's your call.

Reply:
@FredyC Yes, I'll try it out and see where I get with it.

Reply:
I am trying something like this:

```TypeScript
const RenderPropsContext = React.createContext('defaultValue');

interface RenderPropsContextTest {
    assert: boolean;
}

const ContextWithDefaultValue = React.createContext<RenderPropsContextTest>({ assert: true });
const ContextWithoutDefaultValue = React.createContext<RenderPropsContextTest>();

const ContextRenderPropsConsumerComponent = (
    <ContextWithDefaultValue.Consumer>
        {value => value.assert}
    </ContextWithDefaultValue.Consumer>
);

const ContextRenderPropsProviderConsumerComponent = (
    <ContextWithoutDefaultValue.Provider value={{ assert: true }}>
        <ContextWithoutDefaultValue.Consumer>
            {value => value.assert}
        </ContextWithoutDefaultValue.Consumer>
    </ContextWithoutDefaultValue.Provider>
); // $ExpectError
```

Reply:
@FredyC Are these tests sufficient for what you feel should be achieved here?

Reply:
Strange that the error is in the full statement instead of in the argument to `Consumer`, but seems to be the case.

Another _non-erroring_ test where the argument is `value => value === undefined ? null : value.assert` would also demonstrate it.

There's one last confirmation I'd like to take; if the Context is created with a default value, will the Consumer receive that value even if it's not put inside its Provider in the tree? 🤔 _looks at source_